Support for BPMN timer events

CTI777 opened the pull request:
Related to https://github.com/Activiti/Activiti/issues/2636
Depends on 
https://github.com/Activiti/activiti-api/pull/121
https://github.com/Activiti/Activiti/pull/2753
https://github.com/Activiti/activiti-cloud-api/pull/97
https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/322

CTI777 committed:
Timer audit event entities

Following events supported:
TIMER_FIRED
TIMER_SCHEDULED

CTI777 committed:
Timer audit event entities

Following events supported:
TIMER_FIRED
TIMER_SCHEDULED

CTI777 committed:
Some tests added to timer fired/scheduled events

CTI777 committed:
Some tests added to timer fired/scheduled events

CTI777 committed:
Test added for timer event

CTI777 committed:
Test added for timer event

CTI777 committed:
Timer events added

Following events added: 
JOB_CANCELED 
JOB_EXECUTION_SUCCESS 
JOB_EXECUTION_FAILURE 
JOB_RETRIES_DECREMENTED

CTI777 committed:
Timer events added

Following events added: 
JOB_CANCELED 
JOB_EXECUTION_SUCCESS 
JOB_EXECUTION_FAILURE 
JOB_RETRIES_DECREMENTED

ffazzini commented on activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/TimerCanceledEventConverter.java:
```diff
@@ -0,0 +1,59 @@
+import org.activiti.cloud.services.audit.jpa.events.AuditEventEntity;
+import org.activiti.cloud.services.audit.jpa.events.TimerCanceledAuditEventEntity;
+
+public class TimerCanceledEventConverter extends BaseEventToEntityConverter {
```
same suggestion from previous prs about generics might apply here as well?

CTI777 committed:
Refactoring of timer events

CTI777 committed:
Refactoring of timer events

CTI777 committed:
Tests for timer event converters added

CTI777 committed:
Tests for timer event converters added

erdemedeiros commented on activiti-cloud-services-audit-jpa/src/test/java/org/activiti/cloud/services/audit/jpa/controller/AuditEventsControllerImplIT.java:
```diff
@@ -345,4 +349,44 @@ public void getSignalEventById() throws Exception {
     }
+    
+    @Test
+    public void getTimerEventById() throws Exception {
```
Could you rename this test to `shouldGetTimerEventById`?

eromano commented on activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/TimerCancelledEventConverter.java:
```diff
@@ -0,0 +1,59 @@
+    @Override
+    protected TimerCancelledAuditEventEntity createEventEntity(CloudRuntimeEvent cloudRuntimeEvent) {
+        CloudBPMNTimerCancelledEvent cloudEvent = (CloudBPMNTimerCancelledEvent) cloudRuntimeEvent;
+        return new TimerCancelledAuditEventEntity(cloudEvent.getId(),
```
Uncle Bob in the Clean Code book explains that an ideal function is a niladic function (0 arguments)
1 argument monadic function Not ideal
2 arguments dyadic function Not ideal
3 arguments triadic function
More then  3 arguments polyadic function No to be used

Would not be better define an interface as an input parameter and let all the object implement that interface, and do the assignment job only in the super object?

erdemedeiros replied:
Totally agree with Eugenio. All the events are using big constructors, but we need to stop doing so.

CTI777 replied:
I do not like big constructors as well... But I have tried to make everything close / similar to already implemented EventEntities... It means that we have to change all, in principle... Should I apply changes for all eventEntities here, or better to make a new PR and apply changes in one shot?

erdemedeiros replied:
Now we fixed that for timers, but you're right, we need to fix that for other events too. Let's do that in a separated PR.

CTI777 committed:
Merge remote-tracking branch 'origin/CTI777-2636-Add-Support-for-BPMN-Timer-elements' into CTI777-2636-Add-Support-for-BPMN-Timer-elements

CTI777 committed:
Function name changed

erdemedeiros committed:
Start refactoring for AuditEventEntities

- create constructor for AuditEventEntity that takes cloud event as parameter
- move set of common properties to BaseEventToEntityConverter

CTI777 committed:
Refactoring of timer AuditEventEntities

erdemedeiros committed:
Remove unused constructors + fix test
